batch.go

```diff
@@ -36,7 +38,18 @@ type WriteBatch struct {
 // creating and committing transactions. Due to the nature of SSI guaratees provided by Badger,
 // blind writes can never encounter transaction conflicts (ErrConflict).
 func (db *DB) NewWriteBatch() *WriteBatch {
-	return &WriteBatch{db: db, txn: db.newTransaction(true, true)}
+	return &WriteBatch{
+		db:       db,
+		txn:      db.newTransaction(true, true),
+		throttle: y.NewThrottle(100),
+	}
+}
+
+// SetMaxPendingTxns sets a limit on maximum number of pending transactions
+// while writing batches. Default limit is 100. This function should be called
+// at the initialisation of WriteBatch.
+func (wb *WriteBatch) SetMaxPendingTxns(max int) {
+	wb.throttle = y.NewThrottle(max)
```

Comment:
Can this leak memory by any chance? Was reading about timers leaking memory recently, seemed similar. I can't think of reason that it would though.